pipeline/scripts/13-excelExport.R

```diff
@@ -398,39 +398,54 @@ ggsave(paste0(outdir, "/plots/IS_line_select_sum.png"), plot = IS_sum_selection_
 
 ### POSITIVE CONTROLS CHECK
 # these positive controls need to be in the samplesheet, in order to make the Pos_Contr.RData file
-positivecontrol_list <- c('P1002.1_Zscore', 'P1003.1_Zscore', 'P1005.1_Zscore')
+# Positive control samples all have the format P1002.x, P1003.x and P1005.x (where x is a number)
+
+# positivecontrol_list <- c('P1002.1_Zscore', 'P1003.1_Zscore', 'P1005.1_Zscore')
+column_list <- colnames(outlist)
+patterns <- c("^(P1002\\.)[[:digit:]]+_", "^(P1003\\.)[[:digit:]]+_", "^(P1005\\.)[[:digit:]]+_")
+positive_controls_index <- grepl(pattern=paste(patterns, collapse="|"), column_list)
+positivecontrol_list <- column_list[positive_controls_index]
+
 if (z_score == 1) {
-  # find if one or more positive control samples are missing and put in list
+  # find if one or more positive control samples are missing
   missing_pos <- c()
-  for (pos in positivecontrol_list) {
-    if (!(pos %in% colnames(outlist))) {
-      missing_pos <- c(missing_pos, pos)
-    }}
+  # any() grep because you get a vector of FALSE's and TRUE's. only one grep match is needed for each positive control
+  if (any(grep("^(P1002\\.)[[:digit:]]+_", positivecontrol_list)) && 
+      any(grep("^(P1003\\.)[[:digit:]]+_", positivecontrol_list)) && 
+      any(grep("^(P1005\\.)[[:digit:]]+_", positivecontrol_list))){
+    cat("All three positive controls are present")
+  } else {
+    missing_pos <- paste0(c("positive controls list is not complete: ", positivecontrol_list), collapse=" ")
```

Comment:
This comment should be clearer, as the user could interpret it the other way around. As if these are the missing pos controls.

Reply:
Naming should be more intuitive, it used to be the missing pos contrs, but now it's the opposite of missing. Change the name of the variable, warning text, and file name in line 458.

Reply:
changed name of the missing_pos variable and text in e-mail to be more clear